### PR TITLE
Apply scale

### LINKED ISSUE #6: Apply scale parameter
When working with parameters like curvature, I think it is dangerous to have the scale parameter set in the layer properties.  ´This could lead to a sphere appearing perfectly spherical while, in fact, being ellipsoidal.

We could fix this by introducing a small function that "applies" the scale parameter to the points in points/surface layers as this wouldn't change the actual amount of data to be handled - unlike with image data.

### PR DESCRIPTION
**Pull request by jo-mueller**
Fixes #6

### CODE REVIEW

File: napari_process_points_and_surfaces/__init__.py [hunk 1]
```diff
@@ -340,7 +344,10 @@ def points_to_labels(points_data:PointsData, as_large_as_image:LayerData, viewer
 
     # labels_stack[points_data] = np.arange(len(points_data))
 
-    labels_stack = np.zeros(as_large_as_image.shape, dtype=int)
+    labels_stack = np.zeros(as_large_as_image.data.shape, dtype=int)
```
Comment (haesleinhuepf) [inline napari_process_points_and_surfaces/__init__.py:347]: Will this work if `as_large_as_image` is of type `LayerData`?

File: napari_process_points_and_surfaces/_tests/test_function.py [hunk 2]
```diff
@@ -60,11 +64,14 @@ def test_curvature():
...
     add_curvature_scalars(surface_data, Curvature.Minimum_Curvature)
-    
+
     add_spherefitted_curvature(surface_data, radius=1)
-    
\ No newline at end of file
+
+if __name__ == '__main__':
+    import napari
+    test_something(napari.Viewer)
```
Comment (haesleinhuepf) [inline napari_process_points_and_surfaces/_tests/test_function.py:77]:
```suggestion
```

You do not need this if you run `pytest -k test_curvature` from the command line.

File: napari_process_points_and_surfaces/__init__.py [hunk 3]
```diff
@@ -419,40 +426,49 @@ def surface_from_point_cloud_ball_pivoting(points_data:PointsData, radius: float
...
     from skimage.measure import marching_cubes
 
+    if isinstance(labels, Labels):
+        labels = labels.data
+
     binary = np.asarray(labels == label_id)
 
     vertices, faces, normals, values = marching_cubes(binary, 0)
 
+    if isinstance(labels, Labels):
```
Comment (haesleinhuepf) [inline napari_process_points_and_surfaces/__init__.py:448]: I'm suspecting this will never be true because of lines 441/442 above

**Review by haesleinhuepf — CHANGES_REQUESTED** [review]
Hey Johannes @jo-mueller ,

thanks for working on this.

I'm wondering if it is possible to write a bit cleaner code. I suggest, either all functions consume layers or all functions consume nparrays. That would make it easier to maintain later.

I often see syntax like `labels_data: Union[[napari.layers.Labels, napari.types.LabelsData]]` as parameter type annotation. Have you tested this from napari? If you run this function from within napari, how does the user-interface look like? I'm wondering how magicgui can guess how to build a user-interface for this type...

Thanks!
Robert

**Comment by jo-mueller** [comment]
Hey @haesleinhuepf 

> I'm wondering if it is possible to write a bit cleaner code?

Sorry about that. I changed the PR to draft mode for fixing it up and will notify you once it's ready for review.

>I suggest, either all functions consume layers or all functions consume nparrays. That would make it easier to maintain later.

In that case it would have to be layers - otherwise there's no way to access the scale parameter inside a function.

**Pull request closed without merging**